### ISSUE TITLE
VSCodeDebugProtocol: Synchronous operation cancelled!

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#5659]
From customer logs:
7.7 Preview (7.7 build 768)

```
Waiting for communication on stdin/stdout...
ERROR [2018-08-03 12:19:31Z]: An unhandled exception has occurred. Terminating Visual Studio? False
System.OperationCanceledException: Synchronous operation cancelled!
  at Microsoft.VisualStudio.Shared.VSCodeDebugProtocol.Protocol.DebugProtocol.WaitForSyncCompletion (System.Threading.WaitHandle evt) [0x00047] in <b9439701a91a482bb6e98642b96a0c1c>:0 
  at Microsoft.VisualStudio.Shared.VSCodeDebugProtocol.Protocol.DebugProtocol.SendRequestSyncCore[TArgs] (Microsoft.VisualStudio.Shared.VSCodeDebugProtocol.Messages.IDebugRequest`1[TArgs] request) [0x00069] in <b9439701a91a482bb6e98642b96a0c1c>:0 
  at Microsoft.VisualStudio.Shared.VSCodeDebugProtocol.DebugProtocolHost.SendRequestSync[TArgs] (Microsoft.VisualStudio.Shared.VSCodeDebugProtocol.Messages.DebugRequest`1[TArgs] request) [0x00000] in <b9439701a91a482bb6e98642b96a0c1c>:0 
  at MonoDevelop.Debugger.VsCodeDebugProtocol.VSCodeDebuggerSession.Dispose () [0x00026] in /Users/vsts/agent/2.136.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger.VSCodeDebugProtocol/MonoDevelop.Debugger.VsCodeDebugProtocol/VSCodeDebuggerSession.cs:438 
  at DotNetCore.Debugger.DotNetCoreDebuggerSession.Dispose () [0x0000c] in <715aa2ac0eda40099c7c3a955da1de18>:0 
  at MonoDevelop.Debugger.DebuggingService+SessionManager.Dispose () [0x0003b] in /Users/vsts/agent/2.136.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs:810 
  at MonoDevelop.Debugger.DebuggingService+<>c__DisplayClass90_0.<Cleanup>b__1 (System.Threading.Tasks.Task t) [0x00000] in /Users/vsts/agent/2.136.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/DebuggingService.cs:470 
  at System.Threading.Tasks.ContinuationTaskFromTask.InnerInvoke () [0x00024] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/TaskContinuation.cs:63 
  at System.Threading.Tasks.Task.Execute () [0x00010] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/Task.cs:2509 
```

> VS bug [#662499](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/662499)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.2.27 (release-7.6)

Pull Request #5768 merged by: Lluis Sanchez
Author: mono
Commit: 31f4d00c095234bcf2c3e7eaf6c0017f2cf9affa (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.233 (master)

Pull Request #5777 merged by: Lluis Sanchez
Author: mono
Commit: 7da2e660578cfae9bfb177d3b111a7df1dbea980 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1140 (release-7.7)

Pull Request #5777 merged by: Lluis Sanchez
Author: mono
Commit: 7da2e660578cfae9bfb177d3b111a7df1dbea980 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5777 merged by: Lluis Sanchez
Author: mono
Commit: 7da2e660578cfae9bfb177d3b111a7df1dbea980 (xamarin/monodevelop)
